neo/UInt160.cs

```diff
@@ -31,16 +31,19 @@ public UInt160(byte[] value)
         /// Method CompareTo returns 1 if this UInt160 is bigger than other UInt160; -1 if it's smaller; 0 if it's equals
         /// Example: assume this is 01ff00ff00ff00ff00ff00ff00ff00ff00ff00a4, this.CompareTo(02ff00ff00ff00ff00ff00ff00ff00ff00ff00a3) returns 1
         /// </summary>
-        public int CompareTo(UInt160 other)
+        public unsafe int CompareTo(UInt160 other)
         {
-            byte[] x = ToArray();
-            byte[] y = other.ToArray();
-            for (int i = x.Length - 1; i >= 0; i--)
+            fixed (byte* px = ToArray(), py = other.ToArray())
             {
-                if (x[i] > y[i])
-                    return 1;
-                if (x[i] < y[i])
-                    return -1;
+                uint* lpx = (uint*)px;
+                uint* lpy = (uint*)py;
+                for (int i = 5; i >= 0; i--)
```

Comment:
i = 5 or x.Lenght ?

Reply:
Even faster potentially, depending how the compiler optimizes:

```
byte[] xB = ToArray();
fixed (byte* px = xB, py = other.ToArray())
{
    uint *lpxEnd = (uint*)px+xB.Length;
    while (lpx < lpxEnd)
    {
        x = *(lpx++);
        y = *(lpy++);
        if (x > y) return 1;
        if (y > x) return -1;
    }
}
```

Also, since this code is the same for both, can it be in the base class?

Reply:
You're right. I think we can write a pack of unit tests with these three options, and actually run thousands of times and time then on C#. So we decide clearly, and leave this recorded for the future.

Reply:
And this should actually be i=4 because I’d include 0